### ISSUE TITLE
Package installs as 0.0.0 on Python 3.7

### ISSUE THREAD

**Issue by jaraco** [pypa/setuptools-scm#129]
Originally reported as pytest-dev/pytest-runner#23.

Consider this minimal package:

```
$ cat setup.py
import setuptools
__name__ == '__main__' and setuptools.setup(
  name='foo',
  use_scm_version=True,
  setup_requires=['setuptools_scm>0'],
)
```

Installing it under Python 3.7, the version of setuptools_scm isn't detected during install.

```
$ PYTHONPATH=~/m/setuptools ~/p/cpython/python.exe setup.py sdist
/Users/jaraco/p/cpython/Lib/distutils/dist.py:261: UserWarning: Unknown distribution option: 'use_scm_version'
  warnings.warn(msg)

Installed /Users/jaraco/Dropbox/code/main/pytest-runner/issue23/.eggs/setuptools_scm-0.0.0-py3.7.egg
Traceback (most recent call last):
  File "setup.py", line 5, in <module>
    setup_requires=['setuptools_scm>0'],
  File "/Users/jaraco/p/cpython/Lib/distutils/core.py", line 108, in setup
    _setup_distribution = dist = klass(attrs)
  File "/Users/jaraco/m/setuptools/setuptools/dist.py", line 315, in __init__
    self.fetch_build_eggs(attrs['setup_requires'])
  File "/Users/jaraco/m/setuptools/setuptools/dist.py", line 361, in fetch_build_eggs
    replace_conflicting=True,
  File "/Users/jaraco/m/setuptools/pkg_resources/__init__.py", line 853, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'setuptools_scm>0' distribution was not found and is required by the application
```

**Comment by RonnyPfannschmidt** [comment]
there is no python 3.7 egg published, it should be able to fall back to the sdist, whats happening there?

**Comment by RonnyPfannschmidt** [comment]
actually it did fall back to the sdist, i strongly suspect the setuptools sandbox breaks setuptools_scm finding its own metadata (which didnt happen with pip)

on the grand scale of things yet another reason to kill easy_install
its not clear how to fix it nicely
without bringing back the change that made the travis pypi deploy missbehave

the quickest fix is probably uploading a python 3.7 egg (so its not facing users) while working on a more sustainable solution

**Comment by jaraco** [comment]
> its not clear how to fix it nicely without bringing back the change that made the travis pypi deploy missbehave

Maybe it makes sense to to undo this change and write a custom travis pypi deploy step.

> quickest fix is probably uploading a python 3.7 egg

I'm okay with this approach as well, even long term, as it should only emerge with each new Python release. I do wish to deprecate easy_install, which would also address the issue.
